cypress/tests/ui/menu/menu.cy.js

```diff
@@ -1 +1,172 @@
 // All E2E Tests Related to menu goes here.
+
+import { prettifyName } from '../../../../src/utils';
+
+describe('Menu', () => {
+  it('verifies that users can select a section of the flowchart, through the drop down. #TC-16', function () {
+    // Alias
+    cy.intercept('GET', '/api/pipelines/*').as('pipelineRequest');
+    cy.get(':nth-child(2) > .menu-option__content > span').as('menuOption');
+
+    let menuOptionValue;
+
+    cy.get('@menuOption')
+      .invoke('text')
+      .then((menuOptionText) => {
+        menuOptionValue = menuOptionText;
+      });
+
+    // Action
+    cy.get('[data-test="kedro-pipeline-selector"]').click();
+    cy.get('@menuOption').click({ force: true });
+
+    // Adding an extra wait time due to flakiness
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(500);
+
+    // Assert after action
+    cy.wait('@pipelineRequest').then((interception) => {
+      // API Request URL
+      const requestUrl = interception.request.url;
+      expect(decodeURIComponent(requestUrl).toLowerCase()).to.include(
+        `/api/pipelines/${menuOptionValue.toLowerCase()}`
+      );
+
+      // Browser URL
+      cy.url().then((url) => {
```

Comment:
very nice 👍 

I wonder if you can use `cy.location()` instead `cy.url()` to get the search params so you dont have to do the split etc